hyperspy/io_plugins/protochips.py

```diff
@@ -0,0 +1,183 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+import os
+from datetime import datetime
+import warnings
+
+# Plugin characteristics
+# ----------------------
+format_name = 'Protochips'
+description = 'Reads Protochips log files (heating/baising and gas cell)'
+full_support = False
+# Recognised file extension
+file_extensions = ['csv', 'CSV']
+default_extension = 0
+# Reading capabilities
+reads_images = False
+reads_spectrum = False
+reads_spectrum_image = False
+# Writing capabilities
+writes = False
+
+
+def file_reader(filename, *args, **kwds):
+    csv_file = ProtochipsCSV(filename)
+    return protochips_log_reader(csv_file)
+
+
+def protochips_log_reader(csv_file):
+    csvs = []
+    for key in csv_file.logged_quantity_name_list:
+        csvs.append(csv_file.get_dictionary(key))
+    return csvs
+
+
+class ProtochipsCSV(object):
+
+    def __init__(self, filename, header_line_number=10):
+        self.filename = filename
+        self.raw_header = self._read_header(header_line_number)
+        self.column_name = self._read_column_name()
+        self._read_all_metadata_header()
+        self.logged_quantity_name_list = self.column_name[2:]
+
+        self._read_data(header_line_number)
+
+        self.calibration_file = None
+
+    def get_dictionary(self, quantity):
+        return {'data': self._data_dictionary[quantity],
+                'axes': self._get_axes(),
+                'metadata': self._get_metadata(quantity),
+                'original_metadata': {'Protochips_header':
+                                      self._get_original_metadata()},
+                'mapping': self._get_mapping()}
+
+    def _get_original_metadata(self):
+        d = {'Start time': self.start_datetime}
+        d['Time units'] = self.time_units
+        for quantity in self.logged_quantity_name_list:
+            d['%s_units' % quantity] = self._get_quantity_units(quantity)
+        d['User'] = self.user
+        d['Calibration file name'] = self._parse_calibration_file()
+        return d
+
+    def _get_mapping(self):
+        return {'Protochips_header.date': ("General.time", None), }
+
+    def _get_metadata(self, quantity):
+        return {'General': {'original_filename': os.path.split(self.filename)[1],
+                            'title': '%s (%s)' % (quantity,
+                                                  self._get_quantity_units(quantity)),
+                            'user': self.user,
```

Comment:
Either that or we should update the metadata specifications or  a mixture of the two.

Maybe user could be mapped to a new `General.authors`? With the inception of open data, we may also need a `General.doi`.

I see how notes could be useful. So +1 on adding this to the specifications too. notes or comments or something else?

calibration_files certainly does not belong to general. What is it? If it is required in could go into a new `md.Instrument.x.x` and follow well defined and general specifications. This is only worth it if a HyperSpy function is expected to use this information.

I am not sure about what to do with start_time. It's more specific than date and I see that it could be useful for acquisition devices that save each signal in individual files. What do you think that could be the best place for it?